src/main/kotlin/uk/gov/justice/digital/hmpps/prisonersearch/services/SearchOperations.kt

```diff
@@ -31,10 +31,14 @@ fun BoolQueryBuilder.filterWhenPresent(query: String, value: Any?): BoolQueryBui
   value.takeIf {
     when(it) {
       is String -> it.isNotBlank()
+      is List<*> -> it.isNotEmpty()
```

Comment:
Just trying to understand the usage of `*` here... So I think I get why it's not a List<Any> - you're saying it's a typed List but you're not prepared to commit to which type at compile time... right?  But why is it not just a `List<String>` if the singular form must be a `String`?

Reply:
I agree, I'll update that thank you.

Reply:
Ah so the compiler complains when I try to type it with:
`Cannot check for instance of erased type: List<String>` and recommends using `List<*>`

Reply:
Ah ok - so you can't say "I know that this `Any?` is actually a `List<String>`" because the compiler cannot verify it due to type erasure.  Hmmm interesting.